ucp/_libs/tests/test_mem.py

```diff
@@ -0,0 +1,53 @@
+import array
+import io
+import mmap
+
+import pytest
+
+from ucp._libs import ucx_api
+
+builtin_buffers = [
+    b"",
+    b"abcd",
+    array.array("i", []),
+    array.array("i", [0, 1, 2, 3]),
+    array.array("I", [0, 1, 2, 3]),
+    array.array("f", []),
+    array.array("f", [0, 1, 2, 3]),
+    array.array("d", [0, 1, 2, 3]),
+    memoryview(array.array("B", [0, 1, 2, 3, 4, 5])).cast("B", (3, 2)),
+    memoryview(b"abcd"),
+    memoryview(bytearray(b"abcd")),
+    io.BytesIO(b"abcd").getbuffer(),
+    mmap.mmap(-1, 5),
+]
+
+
+def test_alloc():
+    ctx = ucx_api.UCXContext({})
+    mem = ucx_api.UCXMemoryHandle.alloc(ctx, 1024)
+    rkey = mem.pack_rkey()
+    assert rkey is not None
+
+
+@pytest.mark.parametrize("buffer", builtin_buffers)
+def test_map(buffer):
+    ctx = ucx_api.UCXContext({})
+    mem = ucx_api.UCXMemoryHandle.map(ctx, buffer)
+    rkey = mem.pack_rkey()
+    assert rkey is not None
+
+
+def test_ctx_alloc():
+    ctx = ucx_api.UCXContext({})
+    mem = ctx.alloc(1024)
+    rkey = mem.pack_rkey()
+    assert rkey is not None
+
+
+@pytest.mark.parametrize("buffer", builtin_buffers)
+def test_ctx_map(buffer):
+    ctx = ucx_api.UCXContext({})
+    mem = ctx.map(buffer)
+    rkey = mem.pack_rkey()
+    assert rkey is not None
```

Comment:
I think it would be good with a test that pickles the `rkey` and send it to another peer.

Reply:
My old RMA branch has some code like this, I'll see what I can grab from that

Reply:
I think this is the only comment that I didn't completely address. I added a test that will unpack the rkey locally so we know that the rkey is valid and will unpack correctly. I plan on extending `test_peer_send_recv.py` in the next update with tests to the added puts and gets as well as sending the `rkey`.